File: src/main/java/us/illyohs/mod/civilmagiks/common/sigil/nonpersistent/SpeedSigil.java
package us.illyohs.mod.civilmagiks.common.sigil.nonpersistent;

import net.minecraft.entity.Entity;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.potion.Potion;
import net.minecraft.potion.PotionEffect;
import net.minecraft.util.ResourceLocation;
import us.illyohs.mod.civilmagiks.api.sigil.Sigil;

public class SpeedSigil extends Sigil {

    public SpeedSigil(String name, String description, String lore, ResourceLocation texture, int radius,
            int lightLevel, boolean isPersistent, boolean usesMana, boolean canOverlap) {
        super(name, description, lore, texture, radius, lightLevel, isPersistent, usesMana, canOverlap);
    }
    
    public void onEntityCollied(Entity entity) {
        EntityPlayer player = (EntityPlayer)entity;
        if (entity instanceof EntityPlayer) {
            player.addPotionEffect(new PotionEffect(Potion.moveSpeed.id, 4, 5000));
        }
    }
}

File: src/main/scala/us/illyohs/civilmagiks/common/sigil/SpeedSigil.scala
/**
 * Copyright (c) 2016, Anthony Anderson<Illyohs>
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions are met:
 *
 *  Redistributions of source code must retain the above copyright notice, this
 *    list of conditions and the following disclaimer.
 *
 *  Redistributions in binary form must reproduce the above copyright notice,
 *    this list of conditions and the following disclaimer in the documentation
 *    and/or other materials provided with the distribution.
 *
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
 * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
 * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
 * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
 * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
 * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
 * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
 * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
 * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */

package us.illyohs.civilmagiks.common.sigil

import net.minecraft.entity.{Entity, EntityLivingBase}
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.init.MobEffects
import net.minecraft.nbt.NBTTagCompound
import net.minecraft.potion.PotionEffect
import net.minecraft.util.math.BlockPos
import net.minecraft.world.World

import us.illyohs.civilmagiks.common.core.util.SigilBase

class SpeedSigil extends
  SigilBase("speed", 3, 20) {

  override def onUpdate(): Unit = {

  }

  override def onPlayerClick(world: World, pos: BlockPos, player: EntityPlayer): Unit = {
  }

  override def onEntityCollied(world: World, pos: BlockPos, entity: Entity): Unit = {
  }

  override def onEntityLivingCollied(world: World, pos: BlockPos, entity: EntityLivingBase): Unit = {
    entity.addPotionEffect(new PotionEffect(MobEffects.SPEED, 600))
  }

  override def writeToNBT(tag: NBTTagCompound): Unit = {
  }

  override def readFromNBT(tag: NBTTagCompound): Unit = {
  }
}
